src/olympia/blocklist/cron.py

```diff
@@ -108,33 +111,40 @@ def _upload_mlbf_to_remote_settings(*, force_base=False):
 
     statsd.incr(
         'blocklist.cron.upload_mlbf_to_remote_settings.blocked_count',
-        len(mlbf.blocked_items),
+        len(mlbf.data[data_type]),
     )
     statsd.incr(
         'blocklist.cron.upload_mlbf_to_remote_settings.not_blocked_count',
-        len(mlbf.not_blocked_items),
+        len(mlbf.data[MLBFDataType.NOT_BLOCKED]),
     )
 
     make_base_filter = (
         force_base
-        or not base_generation_time
-        or mlbf.blocks_changed_since_previous(base_filter) > BASE_REPLACE_THRESHOLD
+        or base_filter is None
+        or mlbf.blocks_changed_since_previous(data_type, base_filter)
+        > BASE_REPLACE_THRESHOLD
     )
 
-    if previous_filter and not make_base_filter:
+    if not previous_filter or make_base_filter:
         try:
-            mlbf.generate_and_write_stash(previous_filter)
-        except FileNotFoundError:
-            log.info("No previous blocked.json so we can't create a stash.")
-            # fallback to creating a new base if stash fails
-            make_base_filter = True
-    if make_base_filter:
-        mlbf.generate_and_write_filter()
-
-    upload_filter.delay(generation_time, is_base=make_base_filter)
-
-    if base_generation_time:
-        cleanup_old_files.delay(base_filter_id=base_generation_time)
+            mlbf.generate_and_write_filter(data_type)
+        except InvalidErrorRateException:
+            log.warning(
+                'Invalid error rates, because '
+                'all versions are either blocked or not blocked'
+            )
+            return
+    else:
+        mlbf.generate_and_write_stash(data_type, previous_filter)
```

Comment:
for filters generated from DB the cache.json is guaranteed to exist on initialiation so it will raise before this point. for filters loaded from data, if the cache doesn't exist it will raise there and it will be a catastrophic failure.

Reply:
We could consider keeping the try catch but it would have to be somewhere else.

Reply:
I've added a second argument to `load_from_storage` to determine if we want to raise on file not found.

In the cron job if the file isn't found the filter class returns none which is already handled appropriatly. However in the task for uploading, we expect the filter to be found so we raise if it is not.